spec/models/spree/variant_spec.rb

```diff
@@ -23,20 +23,35 @@
     end
 
     describe "tax category" do
-      context "when a tax category is required" do
-        it "is invalid when a tax category is not provided" do
-          with_products_require_tax_category(true) do
-            expect(build_stubbed(:variant, tax_category_id: nil)).not_to be_valid
-          end
-        end
+      # `build_stubbed` avoids creating a tax category in the database.
+      subject(:variant) { build_stubbed(:variant) }
+
+      it "is valid when empty by default" do
+        expect(variant.tax_category).to eq nil
+        expect(variant).to be_valid
       end
 
-      context "when a tax category is not required" do
-        it "is valid when a tax category is not provided" do
-          with_products_require_tax_category(false) do
-            expect(build_stubbed(:variant, tax_category_id: nil)).to be_valid
-          end
-        end
+      it "loads the default tax category" do
+        default = create(:tax_category, is_default: true)
+
+        expect(variant.tax_category).to eq default
+        expect {
+          variant.tax_category = nil
+        }.to_not change {
+          variant.tax_category
+        }
```

Comment:
I am not following what we are testing here ? Are we just checking that a tax category can't be set to null if a default one exists ?

Reply:
> Are we just checking that a tax category can't be set to null if a default one exists ?

Yes, exactly. Well, I don't actually know what would be saved to the database and it doesn't really matter but the getter method always returns the default in this case.